frontend/js/index.js

```
@@ -164,8 +186,112 @@
+      const input = document.createElement('textarea');
+      input.value = originalText;
+      textElement.replaceWith(input);
+      input.focus();
+
+      // Сохраняем изменения
+      input.addEventListener('blur', async () => {
+        const newText = input.value.trim();
+        if (newText && newText !== originalText) {
+          try {
+            await fetch(`http://localhost:4000/messages/${messageElement.dataset.messageId}`, {
+              method: 'PATCH',
+              headers: { 'Content-Type': 'application/json' },
+              body: JSON.stringify({ text: newText })
+            });
+            getMessages();
+            menu.classList.remove('show');
+isMenuOpen = false;
+startPolling();
+          } catch (error) {
+            alert('Error updating message');
+          }
+        } else {
+          textElement.textContent = originalText;
+          input.replaceWith(textElement);
+        }
+      });
+    }
+
+    // Обработка удаления
+    if (deleteBtn) {
+      if (confirm('Delete this message?')) {
+        try {
+          await fetch(`http://localhost:4000/messages/${messageElement.dataset.messageId}`, {
+            method: 'DELETE'
+          });
+          messageElement.remove();
+          menu.classList.remove('show');
+isMenuOpen = false;
+startPolling();
+        } catch (error) {
+          alert('Error deleting message');
+        }
+      }
+    }
+  });
+
+  // Добавьте этот обработчик для закрытия меню при клике вне
+  document.addEventListener('click', (e) => {
+    if (!e.target.closest('.message-menu') && !e.target.closest('.message-menu-btn')) {
+      document.querySelectorAll('.message-menu').forEach(menu => {
+        menu.classList.remove('show');
+        isMenuOpen = false;
+        startPolling();
+      });
+    }
+  });
+
+    
   }
 
+  function startPolling() {
+  if (!messagesInterval) {
+    messagesInterval = setInterval(() => {
+      if (!isMenuOpen) getMessages();
```

Я бы советовал даже если `if` однострочный использовать блоки кода. 

Ну и логика fail fast всегда лучше:
```js
(function() {
  if (isMenuOpen) {
    return;
  }

  getMessages();
});
```

Она лучше, потому что она делает код более плоским. Когда вложенность кода низкая, становится более понятна линейная логика кода.